bugbug/repository.py

```diff
@@ -41,31 +60,85 @@ def __init__(
         desc,
         date,
         pushdate,
-        bug,
+        bug_id,
         backedoutby,
         author_email,
-        files,
-        file_copies,
         reviewers,
     ):
         self.node = node
         self.author = author
         self.desc = desc
         self.date = date
         self.pushdate = pushdate
-        self.bug = bug
+        self.bug_id = bug_id
         self.backedoutby = backedoutby
+        self.ever_backedout = backedoutby != ""
         self.author_email = author_email
-        self.files = files
-        self.file_copies = file_copies
+        self.files = []
+        self.file_copies = {}
         self.reviewers = reviewers
+        self.added = 0
+        self.test_added = 0
+        self.deleted = 0
+        self.test_deleted = 0
+        self.types = set()
+        self.seniority_author = 0
+        self.total_file_size = 0
+        self.average_file_size = 0
+        self.maximum_file_size = 0
+        self.minimum_file_size = 0
+        self.files_modified_num = 0
+        self.components = []
+        self.directories = []
+        self.touched_prev = set()
+
+    def to_dict(self):
+        data = {
+            "node": self.node,
+            "author": self.author,
+            "desc": self.desc,
+            "date": str(self.date),
+            "pushdate": str(self.pushdate),
+            "bug_id": self.bug_id,
+            "ever_backedout": self.ever_backedout,
+            "author_email": self.author_email,
+            "files": self.files,
+            "reviewers": self.reviewers,
+            "added": self.added,
+            "test_added": self.test_added,
+            "deleted": self.deleted,
+            "test_deleted": self.test_deleted,
+            "types": list(self.types),
+            "seniority_author": self.seniority_author,
+            "total_file_size": self.total_file_size,
+            "average_file_size": self.average_file_size,
+            "maximum_file_size": self.maximum_file_size,
+            "minimum_file_size": self.minimum_file_size,
+            "files_modified_num": self.files_modified_num,
+            "components": self.components,
+            "directories": self.directories,
+        }
+        for attr in self.touched_prev:
```

Comment:
No need to store the properties here, you can just iterate through the properties of the object and only consider those which start with "touched_prev".

Reply:
It was to avoid to call __dict__.
But probably we can call __dict__ and then remove what's useless and change type for required keys.